coalib/output/ConsoleInteraction.py

```diff
@@ -922,27 +933,58 @@ def print_bears(bears,
     :param show_params:      True if the parameters and their description
                              should be shown.
     :param console_printer:  Object to print messages on the console.
+    :param args:             Args passed to coala command.
     """
     if not bears:
         console_printer.print('No bears to show. Did you forget to install '
                               'the `coala-bears` package? Try `pip3 install '
                               'coala-bears`.')
         return
 
+    results = []
+    if args and args.json:
+        from coalib.output.JSONEncoder import create_json_encoder
+        JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+
     for bear, sections in sorted(bears.items(),
                                  key=lambda bear_tuple:
                                  bear_tuple[0].name.lower()):
-        show_bear(bear,
-                  show_description,
-                  show_params,
-                  console_printer)
+        if args and (args.json or args.format):
+            results.append(bear)
+        else:
+            show_bear(bear,
+                      show_description,
+                      show_params,
+                      console_printer)
+
+    if args:
+        if args.json:
+            json_output = {'bears': results}
+            import json
+            if args.output:
+                filename = str(args.output[0])
+                with open(filename, 'w+') as fp:
+                    json.dump(json_output, fp,
+                              cls=JSONEncoder,
+                              sort_keys=True,
+                              indent=2,
+                              separators=(',', ': '))
+            else:
+                print(json.dumps(json_output,
+                                 cls=JSONEncoder,
+                                 sort_keys=True,
+                                 indent=2,
+                                 separators=(',', ': ')))
```

Comment:
can't you use `json.dumps` once and just write it to a file? This deduplicates the call ^^